_posts/2025-05-19-journey-to-find-replacements-java-security-manager.md

```diff
@@ -0,0 +1,134 @@
+---
+layout: post
+title: "Finding a replacement for JSM in OpenSearch 3.0"
+authors:
+   - cwperks
+   - kumargu
+   - kolchfa
+date: 2025-05-19
+categories:
+  - technical-posts
+meta_keywords: Opensearch security, Java security manager, OpenSearch 3.0, security, sandboxing.
+meta_description: The Java Security Manager (JSM) has long been a foundational security mechanism in OpenSearch. With its deprecation underway, this blog post explores the alternative technologies adopted to replace JSM.
+---
+
+OpenSearch 3.0.0 introduced many innovative features that provide significant advancements in performance, data management, vector database functionality, and more. In [the release announcement](https://opensearch.org/blog/unveiling-opensearch-3-0/), we shared that OpenSearch has replaced the Java Security Manager (JSM) because of its upcoming deprecation. In this blog post, we share more information about the efforts that enabled this transition.
+
+The decision to remove JSM in 3.0 was carefully considered and primarily driven by two factors:
+
+1. **The upcoming removal of JSM from the Java platform**: JSM has been deprecated since JDK 17 ([JEP411](https://openjdk.org/jeps/411)) and is scheduled for full removal in JDK 24 ([JEP 486](https://openjdk.org/jeps/486)). According to the Java Enhancement Proposal (JEP), this decision was made because very few projects were using JSM, with Elasticsearch being one of the few major open-source projects that did. While OpenSearch 3.0 is bundled with Java 21 and not yet forced to drop JSM, continuing to rely on a deprecated and soon-to-be-removed component was deemed unsustainable for long-term support and innovation.
```

Comment:
```suggestion
1. **The upcoming removal of JSM from the Java platform**: JSM has been deprecated since JDK 17 ([JEP411](https://openjdk.org/jeps/411)) and is scheduled for full removal in JDK 24 ([JEP 486](https://openjdk.org/jeps/486)). According to the Java Enhancement Proposal (JEP), this decision was made because very few projects were using JSM. While OpenSearch 3.0 is bundled with Java 21 and not yet forced to drop JSM, continuing to rely on a deprecated and soon-to-be-removed component was deemed unsustainable for long-term support and innovation.
```